overlord/snapstate/handlers.go

```diff
@@ -898,38 +898,44 @@ func (m *SnapManager) doPreDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 // asyncRefreshOnSnapClose asynchronously waits for the snap the close, notifies
 // the user and then triggers an auto-refresh.
 func asyncRefreshOnSnapClose(st *state.State, refreshInfo *userclient.PendingSnapRefreshInfo) error {
-	var monitoredSnaps map[string]bool
+	var monitoredSnaps map[string]chan<- bool
 	if cachedMonitored := st.Cached("monitored-snaps"); cachedMonitored != nil {
-		monitoredSnaps, _ = cachedMonitored.(map[string]bool)
+		monitoredSnaps, _ = cachedMonitored.(map[string]chan<- bool)
 	} else {
-		monitoredSnaps = make(map[string]bool)
+		monitoredSnaps = make(map[string]chan<- bool)
 	}
 
 	// there's already a goroutine waiting for this snap to close so just notify
-	if monitoredSnaps[refreshInfo.InstanceName] {
+	if monitoredSnaps[refreshInfo.InstanceName] != nil {
 		asyncPendingRefreshNotification(context.TODO(), userclient.New(), refreshInfo)
 		return nil
 	}
 
 	// monitor the snap until it closes
-	done := make(chan string)
+	done := make(chan string, 1)
```

Comment:
this change probably merits a comment, I also wonder in a follow if we want to add a way to actually really stop the monitoring itself?